pkg/discovery/util/pod_util.go

```diff
@@ -424,3 +441,97 @@ func GetPodEvents(kubeClient *client.Clientset, namespace, name string) (podEven
 	}
 	return
 }
+
+// MockPodInterface is used to mock podInterface for testing.
+type MockPodInterface struct {
+	Namespace string
+}
+
+func (p *MockPodInterface) Get(ctx context.Context, name string, opts metav1.GetOptions) (*api.Pod, error) {
+	return nil, nil
+}
+
+func (p *MockPodInterface) List(ctx context.Context, opts metav1.ListOptions) (*api.PodList, error) {
+	pod1 := api.Pod{}
+	pod1.Name = testPodName1
+	pod1.Namespace = p.Namespace
+	pod1.UID = types.UID(testPodUID1)
+	pod1.Status.Phase = api.PodRunning
+
+	pod2 := api.Pod{}
+	pod2.Name = testPodName1
+	pod2.Namespace = p.Namespace
+	pod2.UID = types.UID(testPodUID1)
+	pod2.Status.Phase = api.PodRunning
+
+	podList := api.PodList{
+		Items: []api.Pod{
+			pod1,
+			pod2,
+		},
+	}
+
+	return &podList, nil
+}
+
+func (p *MockPodInterface) Create(ctx context.Context, pod *api.Pod, opts metav1.CreateOptions) (*api.Pod, error) {
+	return nil, nil
+}
+
```

Comment:
Just curious - why are all these test functions in this non-test file?

Reply:
Good question @chlam4. It seems I cannot share code between test files from different packages in golang. [Ref](https://stackoverflow.com/questions/31794141/can-i-create-shared-test-utilities).
I can create another file named something like pod_test_util.go if that sounds better in this case. Please let me know.

Reply:
ic yes then it makes sense to create a different file for the test utils.  Thanks @irfanurrehman!

Reply:
Updated!